.github/workflows/ci.yml

```diff
@@ -62,13 +65,32 @@ jobs:
         run: vagrant ssh ${{env.INSTANCE_NAME}} -c 'sudo make install'
         working-directory: ${{env.BOX_DIR}}
 
+        # NOTE: This special logic with the real block device instead of the loop device is used especially for Fedora 31
+        # due to ussie https://github.com/elastio/elastio-snap/issues/71. It can be simplified back after the issue is fixed.
+      - name: Attach external drive
+        if: ${{ matrix.distro == 'fedora31' }}
+        run: |
+          qemu-img create -f qcow2 ${TEST_IMAGE} 1G
+          virsh attach-disk --domain ${BOX_DIR##*/}_${INSTANCE_NAME} --source ${TEST_IMAGE} --target vdb --driver qemu --subdriver qcow2 --targetbus virtio
+          cd $BOX_DIR && vagrant ssh ${{env.INSTANCE_NAME}} -c 'echo -e "n\np\n\n\n\nw" | sudo fdisk /dev/vdb'
+
       - name: Run tests
-        run: vagrant ssh ${{env.INSTANCE_NAME}} -c 'cd tests && sudo ./elio-test.sh'
+        run: |
+          [[ $INSTANCE_NAME == fedora31* ]] && device=/dev/vdb1
+          vagrant ssh ${{env.INSTANCE_NAME}} -c "cd tests && sudo ./elio-test.sh $device"
         working-directory: ${{env.BOX_DIR}}
         # For now tests are taking 10-20 seconds. But they can hang.
         # 5 minutes seems to be reasonable timeout.
         timeout-minutes: 5
 
+      - name: Detach external drive
+        if: always()
```

Comment:
Isn't `always()` the default if there is no `if` condition?

Reply:
Nope. That's only added in case if we need to run a step even if some previous step has been failed or build is cancelled.
https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#always
The default one is [`if: success()`](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions)